pkg/virt-config/feature-gates.go

```diff
@@ -173,3 +173,7 @@ func (config *ClusterConfig) AlignCPUsEnabled() bool {
 func (config *ClusterConfig) NodeRestrictionEnabled() bool {
 	return config.isFeatureGateEnabled(featuregate.NodeRestrictionGate)
 }
+
+func (config *ClusterConfig) DeclarativeVolumeHotplugEnabled() bool {
+	return config.isFeatureGateEnabled(featuregate.DeclarativeHotplugVolumeGate)
+}
```

Comment:
I wonder if it's safe to infer that declarative hotplug is allowed using the existing hotplug feature gate.

it seems a little awkward that someone would need to enable two different volume hotplug feature gates to toggle slightly different behavior. Especially when i believe the intent behind whether a user wishes to hotplug or not is declared on the volume with the `hotpluggable` boolean.

Reply:
Yeah good question I would be open to that, although I would say we should have removed the `HotplugVolumes` feature gate long ago. I would say the only reason to have two gates is if we intent to get rid of `HotplugVolumes` very soon

Reply:
I suppose the reason we haven't removed the `HotplugVolumes` feature gate yet is that we would like to not use the attachment pods concept for it. But not sure if that will ever happen or not. I guess it is also an implementation detail, and that should not affect if we need a feature gate or not.

Reply:
I tend to think this discussion belongs to the [VEP](https://github.com/kubevirt/enhancements/pull/12) rather the implementation.